homeassistant/components/zha/__init__.py

```diff
@@ -161,3 +152,28 @@ async def async_load_entities(hass: HomeAssistantType) -> None:
         if isinstance(res, Exception):
             _LOGGER.warning("Couldn't setup zha platform: %s", res)
     async_dispatcher_send(hass, SIGNAL_ADD_ENTITIES)
+
+
+async def async_migrate_entry(
+    hass: HomeAssistantType, config_entry: config_entries.ConfigEntry
+):
+    """Migrate old entry."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    if config_entry.version == 1:
+        config_entry.data = {
```

Comment:
The data attribute should be considered read only. We set it to a mapping proxy when the config entry is defined. We should use `hass.config_entries.async_update_entry` to update the data.

Reply:
ok, i'll update it. 
BTW this was copied that from `axis` integration, so that may need update too.

Reply:
ok, it's using `async_update_entry()` now. For config entry migration, i still have to set the `config_entry.version` attribute to the new version, right?